options.js

```diff
@@ -75,88 +75,101 @@ document.getElementById( "switch_to_page_domains_list" ).addEventListener( 'clic
 	} );
 
 // helper function
-function get_active_tab() {
-	return chrome.tabs.query({
+function get_active_tab( callback ) {
+	chrome.tabs.query({
 		// without this will try to do it for all tabs (even without clock (aka not loaded), i guess):
 		active: true,
 		currentWindow: true
-	});
+		}, ( tab ) => {
+			// console.log ("get_active_tab: type = " + typeof( tab ) +
+			// 	"; object keys = " + Object.keys(tab) );
+			// Using callbacks for chrome manifest v2.
+			callback( tab );
+		}
+	);
 }
 
 let clock_visibility = document.getElementById("hide_clock_checkbox");
 set_state_of_hide_clock_checkbox();
 function set_state_of_hide_clock_checkbox() {
-	get_active_tab()
-	.then( function( tab ) {
-		const sended_message = chrome.tabs.sendMessage(
-			tab[0].id,
-			{ visibility: "what" }
-		)
-		sended_message.then( ( response ) => {
-			if( response.response == "hidden" ) {
-				clock_visibility.checked = true;
-			}
-			if( response.response == "shown" ) {
-				clock_visibility.checked = false;
-			}
-		})
-		.catch( function( error ) {
-			console.error( `overlay_clock switch_checkbox send_message error: ${error}` );
-		})
+	get_active_tab( function( tab ) {
+		try {
+			// console.log ("set_state_of_hide_checkbox: tab[0] = " + tab[0] +
+				// "; keys = " + Object.keys( tab[0] ) );
+			const sended_message = chrome.tabs.sendMessage(
+				tab[0].id,
+				{ visibility: "what" },
+				function ( response ) {
+					if( response.response == "hidden" ) {
```

Comment:
Getting an error on Chrome:

```
Error handling response: TypeError: Cannot read properties of undefined (reading 'response')
```

Reply:
Without arrow function, probably it will start to work (recent commit).